replica/database.hh

```diff
@@ -502,7 +502,7 @@ private:
 
     // This field cashes the last truncation time for the table.
     // The master resides in system.truncated table
-    db_clock::time_point _truncated_at = db_clock::time_point::min();
+    std::optional<db_clock::time_point> _truncated_at;
 
```

Comment:
What's the motivation for the patch? You change _truncated_at from something that cannot be uninitialized to something that can be initialized, then check for it. Or is it to match CQL NULL in the truncation table?